cpp/src/arrow/testing/gtest_util.cc

```diff
@@ -238,7 +238,7 @@ std::string ToStringWithMetadata(const T& t, bool show_metadata) {
 }
 
 std::string ToStringWithMetadata(const DataType& t, bool show_metadata) {
-  return t.ToString();
+  return t.ToString(show_metadata);
 }
```

Comment:
Thanks, but this can actually be simplified now. `ToStringWithMetadata` was created because `DataType::ToString` didn't support the `show_metadata` argument. Now that it does, `ToStringWithMetadata` can probably be suppressed and all calls to `ToStringWithMetadata(t, show_metadata)` replaced with `t.ToString(show_metadata)`.